ts/src/renderer/phaser/scenes/GameScene.ts

```diff
@@ -189,15 +191,55 @@ class GameScene extends PhaserScene {
 				map.addTilesetImage(tileset.name, key);
 			}
 		});
+
+		// instantiate the array that will hold the Layer references
+		this.layers = [];
+
 		data.map.layers.forEach((layer) => {
+
+			// floor, 0
+			// floor2, 1
+			// debris, 2 (returns early)
+			// walls, 3
+			// trees, 4
+
 			if (layer.type !== 'tilelayer') {
 				return;
 			}
-			console.log(layer.name);
-			const tilemapLayer = map.createLayer(layer.name, map.tilesets, 0, 0);
-			tilemapLayer.setScale(scaleFactor.x, scaleFactor.y);
+
+			map.createLayer(layer.name, map.tilesets, 0, 0)
+				.setScale(scaleFactor.x, scaleFactor.y)
+				.setName(`map: ${layer.name}`);
+
+			// add to array this.layers
+			this.layers.push(
```

Comment:
can we use `entityLayer` const here, same as above, and omit comments, the code seems pretty much self-explanatory